src/Transport/FutureResponse.cs

```diff
@@ -25,8 +26,13 @@ namespace Apache.NMS.ActiveMQ.Transport
 	/// <summary>
 	/// Handles asynchronous responses
 	/// </summary>
-	public class FutureResponse
+	public class FutureResponse : TaskCompletionSource<Response>
```

Comment:
So it seems you have 2 `TCS`s to handle a single request-reply. One should be enough.

Reply:
seemed so : ), did a little refactoring, its bit more lean now

Reply:
@Havret, so once you have some time please look again at this pr